plugins/out_cloudwatch_logs/cloudwatch_api.c

```diff
@@ -837,6 +837,13 @@ int process_and_send(struct flb_cloudwatch *ctx, const char *input_plugin,
         /* unpack the array of [timestamp, map] */
         flb_time_pop_from_msgpack(&tms, &result, &obj);
 
+        /* Check type of msgpack object */
+        if (root.via.array.ptr[1].type != MSGPACK_OBJECT_MAP) {
+            flb_plg_error(ctx->ins, "Could not find log_key '%s' in record",
+                ctx->log_key);
+            continue;
+        }
+
         /* Get the record/map */
```

Comment:
This check isn't inside of the log_key if block, so I think it needs to check if log key is configured

Reply:
Wow. Right. This pr would actually cause a lot of damage if published as is... I'll add the check.

Reply:
you should always do some manual regression testing before putting up a PR, like run the test case with the feature off

Reply:
Right, I did the testing only with log_key on. More thoroughly testing is a good idea.

Reply:
### Conf
```
[SERVICE]
     Grace 30
     Log_Level debug

# Provide entry point for logs
[INPUT]
     Name http
     host 0.0.0.0
     port 8888

[OUTPUT]
     Name cloudwatch_logs
     Match *
     log_stream_prefix test
     log_group_name fluent_replay_all
     auto_create_group true
     region us-west-2
```

### Problem
![problem](https://user-images.githubusercontent.com/34408404/135942133-edaae6d0-e986-47fc-a79a-5afa98c3d30c.png)
![result_problem](https://user-images.githubusercontent.com/34408404/135942026-946ba16e-8141-46d7-987c-ca54b708271a.png)

### Resolved
![resolved](https://user-images.githubusercontent.com/34408404/135942133-edaae6d0-e986-47fc-a79a-5afa98c3d30c.png)
![result_resolved](https://user-images.githubusercontent.com/34408404/135942412-b2a87c75-5479-4822-a39d-d295204593cc.png)